src/lib.rs

```diff
@@ -206,7 +216,28 @@ impl<S: Read + Write> Future for ConnectAsync<S> {
     type Error = HandshakeError<S>;
 
     fn poll(&mut self) -> Poll<SslStream<S>, HandshakeError<S>> {
-        self.inner.poll()
+        match mem::replace(&mut self.0, ConnectAsyncInner::Done) {
+            ConnectAsyncInner::Start { config, domain, stream } => {
+                match config.connect(&domain, stream) {
+                    Ok(inner) => Ok(Async::Ready(SslStream { inner })),
+                    Err(HandshakeError::WouldBlock(s)) => {
+                        self.0 = ConnectAsyncInner::Handshake(s);
+                        Ok(Async::NotReady)
+                    }
+                    Err(e) => Err(e),
+                }
+            }
+            ConnectAsyncInner::Handshake(s) => match s.handshake() {
+                Ok(inner) => Ok(Async::Ready(SslStream { inner })),
+                Err(HandshakeError::WouldBlock(s)) => {
+                    self.0 = ConnectAsyncInner::Handshake(s);
+                    Ok(Async::NotReady)
+                }
+                Err(e) => Err(e),
+            }
+            ConnectAsyncInner::Error(e) => Err(e),
```

Comment:
I think this state may no longer be construcated anywhere?

Reply:
It can be if the call to `connector.configure()` fails in the `SslConnectorExt` implementation.